roles/vscode/tasks/main.yml

```diff
@@ -0,0 +1,35 @@
+---
+# tasks file for vscode
+- name: Add Microsoft key
+  apt_key:
+    id: BC528686B50D79E339D3721CEB3E94ADBE1229CF
+    url: https://packages.microsoft.com/keys/microsoft.asc
+    state: present
+- name: Install VSCode Repository file
```

Comment:
Is there a particular preference for this implementation over the following?

```
- apt_repository:
    repo: deb https://packages.microsoft.com/repos/vscode stable main
    state: present
    filename: vscode
```

Reply:
FWIW, an `apt update` is automatically performed when an `apt_repository` entry changes by default.

Reply:
Stole this from robots, but there we're using a template to set the Ubuntu version, so I guess it's less necessary here. And I didn't realize apt-repository was a thing.

Reply:
Yeah, I am a little curious if we should evaluate our use of .list files. Think in most the cases we could use the variable expansion except maybe common where we explicitly want to override entire files... But resolving this in other places can be a separate issue at a later time.